anomdetect/src/test/java/com/expedia/adaptivealerting/anomdetect/source/data/graphite/GraphiteSourceTest.java

```diff
@@ -0,0 +1,100 @@
+package com.expedia.adaptivealerting.anomdetect.source.data.graphite;
+
+import com.expedia.adaptivealerting.anomdetect.source.data.DataSourceResult;
+import lombok.val;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.expedia.adaptivealerting.anomdetect.source.data.graphite.GraphiteClient.DEFAULT_FROM_VALUE;
+import static com.expedia.adaptivealerting.anomdetect.source.data.graphite.GraphiteClient.DEFAULT_MAX_DATA_POINTS;
+import static org.junit.Assert.assertEquals;
+
+import static org.mockito.Mockito.when;
+
+public class GraphiteSourceTest {
+    @Mock
+    private GraphiteClient graphiteClient;
+
+    private GraphiteSource sourceUnderTest;
+    private GraphiteResult[] graphiteResults = new GraphiteResult[2];
+    private GraphiteResult[] graphiteResults_null = new GraphiteResult[2];
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        initTestObjects();
+        initDependencies();
+        this.sourceUnderTest = new GraphiteSource(graphiteClient);
+    }
+
+    @Test
+    public void testGetMetricData() {
+        val dataSourceResult = getDataSourceResult(1.0, 1578307488);
+        DataSourceResult dataSourceResult1 = getDataSourceResult(3.0, 1578307489);
+
+        List<DataSourceResult> dataSourceResults = new ArrayList<>();
+        dataSourceResults.add(dataSourceResult);
```

Comment:
Suggest inlining `dataSourceResult` and `dataSourceResult1` variables for readability:

```
        dataSourceResults.add(getDataSourceResult(1.0, 1578307488));
        dataSourceResults.add(getDataSourceResult(3.0, 1578307489));
```